### PR TITLE
Experimental Reverse Engineering.

### PR DESCRIPTION
**Pull request by Adrian16199**
TO-DO:

- [X] Get a sprite. (Good enough until someone complains):
- [X] Ensure that I dont break the system itself (good enough)
- [X] Get the radiation to work
- [X] Get scanning to become change-able via yml. 
- [X] Make it researchable
Optional:
- [X] Danger Aversion score also being change-able via yml.

Make more stuff RE like syndie magboots RE, esword, bluh bluh, also maybe even somethin that DOESNT require syndie:
SALVAGE:
- [ ] Crushers
SYNDICATE:
- [X] SA-123 magboots
- [X] SA-124 medical hardsuit
- [ ] SA-125 elite hardsuit
- [ ] somethin for esword
- [ ] hypo pen into hyposprays that borgs have.
ENGINEER:
- [ ] RCD
- [ ] RCD ammo
RANDOM STUFF:
Skateboots?

## About the PR

Adds Experimental Reverse Engineering variation with improved safety and speed.
Additional content inform of more special content only obtainable thro reverse engineering.
## Why / Balance

Because I got bored.
## Media

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Added an experimental reverse engineering variant with additional content to reverse engineer.

### CODE REVIEW

**Comment by deltanedas** [comment]
re should have an actual purpose i dont see anyone making this
outside of a syndie specifically giving epi a hardsuit to reverse engineer (which appened before)

**Comment by Adrian16199** [comment]
> re should have an actual purpose i dont see anyone making this outside of a syndie specifically giving epi a hardsuit to reverse engineer (which appened before)

I mean, im listening for suggestions.
Im aware that its not used alot and im simply trying to make an upgraded version of it. Learning on the way aswell.
There could be an argument to allow somethin like captain's hardsuit, HoS's hardsuit, just alot of hardsuits be reverse-engineerable.

**Comment by UnicornOnLSD** [comment]
the list of items that would have an interest to be REd is not large, on the contrary, other than just adding extra contraband, maybe permitting salvage valuables to be REd would be great, namely specific kinetic weapons etc.

**Comment by DangerRevolution** [comment]
only two idea guy suggestions i've got for you is uh; incentivsing Salvage to bring any machines & tools to Epi for them to reverse-engineer. They already can but, not sure what more can be done to incentivise it.

*or*

C# Intensive but uh; something like the upgrader from the SCP games, you stick an item in and you have a chance to upgrade it or destroy it. that type of thing

**Comment by UnicornOnLSD** [comment]
considering hardsuits are basically *non craftable* and the only ones are REable, not certain on that behalf, but hey if raw content is wanted, and we got blukies, could always put some more stuff on the table, esword etc etc.
Would we want to keep the RE machine as purely what it is, that's how I see it's limits

**Comment by Adrian16199** [comment]
> only two idea guy suggestions i've got for you is uh; incentivsing Salvage to bring any machines & tools to Epi for them to reverse-engineer. They already can but, not sure what more can be done to incentivise it.
> 
> _or_
> 
> C# Intensive but uh; something like the upgrader from the SCP games, you stick an item in and you have a chance to upgrade it or destroy it. that type of thing

the second thing..... that sounds like a really cool idea tbh.
I could try that one to learn xaml but probably in a separate PR

**Comment by Adrian16199** [comment]
> considering hardsuits are basically _non craftable_ and the only ones are REable, not certain on that behalf, but hey if raw content is wanted, and we got blukies, could always put some more stuff on the table, esword etc etc. Would we want to keep the RE machine as purely what it is, that's how I see it's limits

Hm, how would an esword be turned into.... our own version. blood-red turns into a blukie for it to be usefull for the crew and not get mistaken, how would an esword be.... hmm

**Comment by DangerRevolution** [comment]
> > considering hardsuits are basically _non craftable_ and the only ones are REable, not certain on that behalf, but hey if raw content is wanted, and we got blukies, could always put some more stuff on the table, esword etc etc. Would we want to keep the RE machine as purely what it is, that's how I see it's limits
> 
> Hm, how would an esword be turned into.... our own version. blood-red turns into a blukie for it to be usefull for the crew and not get mistaken, how would an esword be.... hmm

while SoP does allow for Mysta to use contraband for research purposes; it's rarely done so and so as a result if the Crew gets Bluekies it's specifically a clever Mysta who intercepts the Syndie surrendering their contraband willingly (and probably their ability to play Antag ever again) to the crew

**Comment by Squishy77** [comment]
my idea for the future of the re-machine is to not have like weapons, but eshields and hypos are ok! they both go to the Secfab, but can be accessed from an emaged prolate, so metas can reasonably form around the machine. adding things like hyper capacity batteries and ancient power cells would be great from the salv loot pool. another question to ask should rd servers be re-able. also, unicorn makes a good point about salv kentic weapons I could look into that the next update that would be a good side grade to the machine.

**Comment by deltanedas** [comment]
esword turns into brand safe Photonic Blade :trollface:

**Comment by Adrian16199** [comment]
Mfw the radiation part literly worked the entire time but I literly didnt tested it properly.
Fucking hilarious.

**Comment by UnicornOnLSD** [comment]
>esword turns into brand safe Photonic Blade :trollface:

doing a RE on the sword lets you make toy sword :trollface:

File: Content.Server/Nyanotrasen/ReverseEngineering/ReverseEngineeringMachineComponent.cs [hunk 1]
```diff
@@ -22,6 +22,7 @@ public sealed partial class ReverseEngineeringMachineComponent : Component
     /// <summary>
     /// Added to the 3d6, scales off of scanner.
     /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
```
Comment (deltanedas) [inline Content.Server/Nyanotrasen/ReverseEngineering/ReverseEngineeringMachineComponent.cs]: just do `[DataField]` for all these its automatically readwrite

**Comment by Adrian16199** [comment]
So this is gonna be pain to redo the entire PR just because of github being github.

**Comment by deltanedas** [comment]
the code shouldnt need any changes besides radiation support the safety etc can all be specified in yml now

**Comment by Adrian16199** [comment]
> the code shouldnt need any changes besides radiation support the safety etc can all be specified in yml now

I mean specificly that I cant resolve the conflict here and I have never bothered with git and how unreliable it was to be able to resolve it. Either someone else does it or i have to copy and paste into a new branch.

**Comment by deltanedas** [comment]
you just `git pull <deltav remove> master` and resolve conflicts in the listed files

**Pull request closed without merging**